### ISSUE TITLE
9.2.2.1 Zeitbegrenzungen: WCAG erlaubt Time Out erst nach > 20 Stunden, nicht 20 Minuten

### ISSUE THREAD

**Issue by johannesFischer84** [BIK-BITV/BIK-Web-Test#47]
Abschnitt Prüfung, Nummer 2.2.2, 20-Minuten-Test: Nach den WCAG sind eigentlich 20 Stunden als Ausnahme vorgesehen, nicht 20 Minuten.

Mir ist klar, dass 20 Stunden nicht oder nur sehr umständlich testbar sind. Dennoch finde ich die starre Festlegung auf 20 Minuten nicht korrekt. Vielleicht könnte man in den Hinweisen dazu schreiben, dass in den WCAG 20 Stunden gefordert werden und in der Prüfung je nach Einschätzung des Prüfers das untersuchte Zeitintervall auch länger als 20 Minuten gesetzt werden darf, z. B. bei Formularen mit sehr vielen Eingabemöglichkeiten.

**Comment by sweckenmann** [comment]
@johannesFischer84 Guter Hinweis. Wollte auch schon ein issue anlegen. Danke. In den letzten Tests kam es ab und zu vor, dass ein Timeout erst nach mehr als 20min einsetzte. Laut PS-Beschreibung konnte man kein fail geben, was aber nicht mit der WCAG konform ist. Sofern ein Timeout beim Prüfen auftritt, ist er eigentlich relevant.

**Comment by mayerhh** [comment]
Bei so einer langen Zeitdauer müssten wir dann aber auch berücksichtigen, ob nach dem Timeout und Authentisierung der Prozess ohne Verluste fortgeführt werden kann.  Dann ist die Zeitbegrenzung wohl kein Problem mehr wenn ich das hier richtig verstehe:  https://www.w3.org/WAI/WCAG21/Understanding/timeouts.html

**Comment by johannesFischer84** [comment]
@mayerhh Der Link führt ja mit 2.2.6 auf ein AAA-Kriterium, welches aus meiner Sicht aussagt: Wenn es einen Timeout vor der 20-Stunden-Zeitdauer gibt und es könnten Daten verloren gehen, dann muss der Nutzer gewarnt werden.

Wenn ein Timeout nach mehr als 20 Stunden auftritt, dann wären sowohl 2.2.1 als auch 2.2.6 konform.
Wenn ein Timeout bis 20 Stunden aufritt, und kein Datenverlust eintreten kann, ist 2.2.6 konform, bei 2.2.1 müssten weitere mögliche Ausnahmen (z. B. 10fache Verlängerbarkeit usw.) geprüft werden. Ob tatsächlich kein Datenverlust eintreten kann, wäre bei 2.2.6 zu prüfen.
Wenn ein Timeout bis 20 Stunden aufritt, und Datenverlust eintreten kann, ist 2.2.6 konform, falls der Nutzer gewarnt wird. Bei 2.2.1 müssten weitere mögliche Ausnahmen (z. B. 10fache Verlängerbarkeit usw.) geprüft werden.

Aber 2.2.6 wird meist nicht geprüft, weil AAA meist nicht geprüft wird.

**Comment by MarcHaunschild** [comment]
Schafft es denn jemand einen Test an einem Tag durchzuführen? Von Feierabend des einen Tages bis Arbeitsbeginn des nächsten Tages vergehen weniger als 20 Stunden, daher finde ich die Überprüfung meistens unproblematisch. In der Praxis ist man dann auch rausgeflogen. Ich habe noch kein Angebot gesehen, aus dem man nach 19:59 rausfliegt. Meistens findet ein automatischer Logout nach 1 bis 4 Stunden statt.
